dsp-value-objects/src/main/scala/dsp/valueobjects/V2.scala

```diff
@@ -0,0 +1,371 @@
+   *
+   * @param s        the string to be checked.
+   * @param errorFun a function that throws an exception. It will be called if the string does not represent a valid
+   *                 IRI.
+   * @return the same string.
+   */
+  def validateAndEscapeIri(s: String, errorFun: => Nothing): IRI = {
+    val urlEncodedStr = encodeAllowEscapes(s)
+
+    if (urlValidator.isValid(urlEncodedStr)) {
+      urlEncodedStr
+    } else {
+      errorFun
+    }
+  }
+
+  /**
+   * Check that the supplied IRI represents a valid project IRI.
+   *
+   * @param iri      the string to be checked.
+   * @param errorFun a function that throws an exception. It will be called if the string does not represent a valid
+   *                 project IRI.
+   * @return the same string but escaped.
+   */
+  def validateAndEscapeProjectIri(iri: IRI, errorFun: => Nothing): IRI =
+    if (isKnoraProjectIriStr(iri)) {
+      toSparqlEncodedString(iri, errorFun)
+    } else {
+      errorFun
+    }
+
+  /**
+   * Check that the supplied IRI represents a valid user IRI.
+   *
+   * @param iri      the string to be checked.
+   * @param errorFun a function that throws an exception. It will be called if the string does not represent a valid
+   *                 user IRI.
+   * @return the same string but escaped.
+   */
+  def validateAndEscapeUserIri(iri: IRI, errorFun: => Nothing): String =
+    if (isKnoraUserIriStr(iri)) {
+      toSparqlEncodedString(iri, errorFun)
+    } else {
+      errorFun
+    }
+}
+
+object V2UuidValidation {
+
+  /**
+   * The length of the canonical representation of a UUID.
+   */
+  val CanonicalUuidLength = 36
+
+  /**
+   * The length of a Base64-encoded UUID.
+   */
+  val Base64UuidLength = 22
+
+  /**
+   * Checks if a string is the right length to be a canonical or Base64-encoded UUID.
+   *
+   * @param s the string to check.
+   * @return TRUE if the string is the right length to be a canonical or Base64-encoded UUID.
+   */
+  def hasUUIDLength(s: String): Boolean =
+    s.length == CanonicalUuidLength || s.length == Base64UuidLength
+
+  /**
+   * Checks if UUID used to create IRI has correct version (4 and 5 are allowed).
+   * @param s the string (IRI) to be checked.
+   * @return TRUE for correct versions, FALSE for incorrect.
+   */
+  def isUUIDVersion4Or5(s: String): Boolean =
+    if (getUUIDVersion(s) == 4 || getUUIDVersion(s) == 5) {
+      true
+    } else {
+      false
+    }
+
+  /**
+   * Gets the last segment of IRI, decodes UUID and gets the version.
+   * @param s the string (IRI) to be checked.
+   * @return UUID version.
+   */
+  def getUUIDVersion(s: String): Int = {
+    val encodedUUID = s.split("/").last
+    decodeUuid(encodedUUID).version()
+  }
+
+  /**
+   * Calls `decodeUuidWithErr`, throwing [[InconsistentRepositoryDataException]] if the string cannot be parsed.
+   */
+  def decodeUuid(uuidStr: String): UUID =
+    decodeUuidWithErr(uuidStr, throw V2.BadRequestException(s"Invalid UUID: $uuidStr"))
```

Comment:
This probably should use a pre-prepared error message too, right?

Reply:
It's copied from StringFormatter, it will be refactored in next step.